File: fixtures/scale/ruby/large/lib/check_gronsfeld.rb
require_relative 'gronsfeld_encrypt'
require_relative 'gronsfeld_decrypt'
require_relative 'corpus_additive'

def check_gronsfeld
  corpus_additive.each do |text|
    sealed = gronsfeld_encrypt(text, 4)
    opened = gronsfeld_decrypt(sealed, 4)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_gronsfeld.py
from cipherlab.ciphers.gronsfeld import gronsfeld_encrypt, gronsfeld_decrypt
from cipherlab.specs.gronsfeld_spec import gronsfeld_spec
from cipherlab.corpus.corpus_additive import corpus_additive


def check_gronsfeld():
    spec = gronsfeld_spec()
    for text in corpus_additive():
        sealed = gronsfeld_encrypt(text, spec["key"])
        opened = gronsfeld_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "additive"
